src/main/scala/io/iohk/ethereum/timing/Beacon.scala

```diff
@@ -0,0 +1,54 @@
+package io.iohk.ethereum.timing
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Scheduler}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+
+class Beacon(
+  miner: ActorRef,
+  slotDuration: FiniteDuration,
+  startingTime: Long = System.currentTimeMillis,
+  externalSchedulerOpt: Option[Scheduler] = None)
+  extends Actor
+    with ActorLogging {
+
+  import Beacon._
+
+  val startingSlotNumber: SlotNumber = 1
+
+  def scheduler: Scheduler = externalSchedulerOpt getOrElse context.system.scheduler
+
+  override def receive: Receive = idle
+
+  def idle: Receive = {
+    case Start =>
+      context become start
+      self ! NewSlot(startingSlotNumber)
```

Comment:
What happens if this client is started after `n` slots have passed? Shouldn't the starting slot number be calculated based on the current time and the initial time?
If the fix is too complicated we could document it and leave if for later.

Reply:
I'm not sure about that due to my limited knowledge about blockchain systems :), but if we want to take that scenario into account then yes, we should implement the fix. I don't think it's too complicated so I'll give it a try.

Reply:
Done.